src/Xamarin.Android.Build.Tasks/Utilities/Files.cs

```diff
@@ -211,18 +211,26 @@ public static ZipArchive ReadZipFile (string filename, bool strictConsistencyChe
 			bool updated = false;
 			HashSet<string> files = new HashSet<string> ();
 			foreach (var entry in zip) {
+				progressCallback?.Invoke (i++, total);
 				if (entry.FullName.Contains ("/__MACOSX/") ||
 						entry.FullName.EndsWith ("/__MACOSX", StringComparison.OrdinalIgnoreCase) ||
 						entry.FullName.EndsWith ("/.DS_Store", StringComparison.OrdinalIgnoreCase))
 					continue;
 				var fullName = modifyCallback?.Invoke (entry.FullName) ?? entry.FullName;
 				if (entry.IsDirectory) {
-					Directory.CreateDirectory (Path.Combine (destination, fullName));
+					try {
+						Directory.CreateDirectory (Path.Combine (destination, fullName));
+					} catch (NotSupportedException) {
+						//NOTE: invalid paths, such as `:` on Windows can cause this
+					}
+					continue;
```

Comment:
So I'm not 100% positive that silently failing and continuing is the best option here.

I thought about making it fail and report the path, but in this case it would be just as cryptic:
```
Encountered an invalid path extracting "__AndroidLibraryProjects__.zip":
    library_project_importsC:/projects/pclcrypto/src/.obj/pclcrypto/Release/monoandroid23/jl/assets/
```

I'm open to suggestions!

Reply:
You're absolutely correct that ignoring the exception is inappropriate.

The problem is that the binding assembly is *effectively* invalid: the paths contained within it are bad, and we should feel bad for ever having generated them in the first place.

*At minimum* we need to determine *how* we're generating this invalid `__AndroidLibraryProjects__.zip` file; I think [you're starting to understand this](https://github.com/xamarin/xamarin-android/pull/1852#issuecomment-398471804).

Once we've figured out how to stop producing bizarro `__AndroidLibraryProjects__.zip` files, then we can examine what to do next. I can see only two things to do:

1. Do nothing, and let these things error out as they currently do. We'd then need to ask the people who have built the "offending" assemblies to rebuild them against a fixed toolchain.
2. Try to "fixup" the paths that we get from the `.zip` and do something "sane" with them.

I suspect that the fixes from PR #1788 may make (2) sane-is.

Reply:
I just was able to repro locally in a new unit test.

Looking at this method: https://github.com/xamarin/xamarin-android/blob/ea7441594003c1f8e3b2dcff8f5610bf86e67868/src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs#L87-L102

If I pass in `folder` as:
```
"C:\Users\myuser\Desktop\Git\xamarin-android\bin\TestDebug\temp\../temp/"
```

If `internalDir` evaluates to:
```
"C:\Users\myuser\Desktop\Git\xamarin-android\bin\TestDebug\temp\.temp/B"
```

Then `fullDirPath` is:
```
"tempC:\Users\myuser\Desktop\Git\xamarin-android\bin\TestDebug\temp\.temp/B"
```